src/Sparrow.Server/Platform/RavenProcess.cs

```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using Sparrow.Logging;
+
+namespace Sparrow.Server.Platform
+{
+    public class ProcessExitedEventArgs : EventArgs
+    {
+        public int ExitCode { get; set; }
+        public IntPtr Pid { get; set; }
+    }
+
+    public class RavenProcess : IDisposable
+    {
+        public ProcessStartInfo StartInfo { get; set; }
+        private bool _hasExited;
+
+        public event EventHandler ProcessExited;
+        private void OnProcessExited(EventArgs e)
+        {
+            EventHandler handler = ProcessExited;
+            handler?.Invoke(this, e);
+        }
+
+        public event Action<object, string> LineOutput;
+        private void OnLineOutput(string line, CancellationToken ctk)
+        {
+            if (ctk.IsCancellationRequested == false)
+            {
+                var handler = LineOutput;
+                handler?.Invoke(this, line);
+            }
+        }
+
+        public delegate void StreamWriteDelegate(Span<byte> bytes);
+
+        private readonly Logger _logger = LoggingSource.Instance.GetLogger<RavenProcess>("RavenProcess");
+        private IntPtr _pid = IntPtr.Zero;
+        private SafeFileHandle StandardOutAndErr { get; set; }
+        private SafeFileHandle StandardIn { get; set; }
+
+        public void Start(CancellationToken ctk)
```

Comment:
Can we drop `ctk` parameter? I don't see any usage of it

Reply:
But future usage possible and also makes sense.

Reply:
This is internal API, we can change it at will

Reply:
https://github.com/ravendb/ravendb/pull/9723/commits/35cf7d7b67f312e093d3a1b6ea29fd493c82c10f